spring-cloud-security/src/main/java/org/springframework/cloud/security/oauth2/client/feign/OAuth2FeignRequestInterceptor.java

```diff
@@ -101,6 +102,7 @@ public OAuth2FeignRequestInterceptor(OAuth2ClientContext oAuth2ClientContext,
 	 */
 	@Override
 	public void apply(RequestTemplate template) {
+		template.header(header);
```

Comment:
Explain a bit more why `template.header()` is called twice?

Reply:
That's because there is no possibility to "unset" a header. Calling `template.header(header, extract(tokenType));` will either create the header, or append the value if it was already set.

Reply:
Clarified with [256addc](https://github.com/spring-cloud/spring-cloud-security/pull/216/commits/256addcebc8cae51eeecbf403e2c8f02776a8b94)